exec.js

```diff
@@ -12,29 +12,106 @@ var run = function() {
     });
 };
 
-exports.bindButtons = function () {
-    document.getElementById("ooni_button").onclick = function(){
-        args = document.getElementById("args").value;
-        run();
-    }
-
-    document.getElementById("file_button").onclick = function(){
-        fileName = dialog.showOpenDialog({ properties: [ 'openFile']});
-    }
-}
-
 exports.bindPluginGrid = function(state){
     list = document.getElementsByClassName("plugin");
     for (var i = 0; i < list.length; i++) { // save us ES6!
         list[i].addEventListener('click', function(object){
             state.run.set("selected", object.currentTarget.id);
             toggleOverlay();
+            bindRunButton(state);
         }, true);
     }
 }
 
+var bindRunButton = function(state) {
+    document.getElementById("run-button").addEventListener("click", 
+        function() { 
+            if (formValid())
+                collectArgs(state) 
+        });
+};
+
+/* Reads arguments from the modal and generates a command
+ * to be run. Uses the template from plugins[x].command
+ * and replaces flags as appropriate */
+var collectArgs = function(state) {
+    var containers = document.getElementsByClassName("argument-container");
+    var args = [];
+    var selectedIndex = state.run.get("selected");
+
+    for(var i = 0; i < containers.length; i++) {
+        // activation refers to a checkbox that signifies if the user has activated
+        // the argument (only for optional arguments)
+        var activation = containers[i].getElementsByClassName("activation")[0];
+        if (activation && activation.checked || !activation) {
+            var option = containers[i].getElementsByTagName("input");
+            // to handle dialog boxes
+            if (option.length == 1 && activation) {
+                var option = containers[i].getElementsByTagName("select")[0];
+            }
+            // to handle radio buttons
+            else if (option.length > 2) {
+                for(var k = 0; k < option.length; k++) {
+                    if (option[k].checked && k != 0) {
+                        var option = option[k]
+                        break;
+                    }
+                }
+            }
+            else
+            // for all other inputs, relevant input is always the last one
+                option = option[option.length - 1];
```

Comment:
I guess we can merge the pull request, which overall looks great, and think at this later. I will merge the pull request later this afternoon and add an issue to remind us of this.